### ISSUE TITLE
replication on iOS + sqlite plugin

### ISSUE THREAD

**Issue by finchen** [pouchdb/pouchdb#4527]
Hi,

I have a weird error when I run the following code in the web inspector on an iOS 9 emulator on a mac
pouchdb: 4.0.7
couchdb: 1.6.0

```javascript
// local sqlite db (cordova plugin 0.7.11)
var db = new PouchDB('test', {adapter: 'websql'});
// remote 20mb database
var remoteDB = new PouchDB("http://xxxxx:5984/mydb",{ajax: {timeout: 60000}});

// replication is a success
db.replicate.from(remoteDB)
.on('active', function () {
     console.log('Replication started');
})
on('complete', function (res) {
    console.log("DB synchronized!", res);
})
.on('error', function (err) {
        console.log('Replication error', err);
});

// but a query
db.allDocs({
                  include_docs: true,
                  attachments: false
                }).then(function(doc){
                        console.log(doc)
                });
// = SyntaxError: Unexpected EOF http://192.168.1.4/#/home
```

instead of the replication, if I insert a *single doc*, it is working.
```javascript
db.put({
  _id: 'mydoc',
  type: 'type1',
  category: 'category1',
  title: 'Heroes'
});
```
if I load with pouchdb-load instead of the replication, I get the same error

if I remove the sqlite plugin, it is working. 
Lucky my database is still under 50mb, but I am a bit worry about the future.

what is your advise to debug?
should I try your debug sqlite plugin?

**Comment by finchen** [comment]
After resetting the ios emulator 'Reset Content and Settings'
and loading the database from pouchdb-load at the first use, it is working fine with sqlite.
(index, queries, etc) 

At the first sync, it starts but does not complete.
Then it is back with an error but the console display fails and shows an EOF syntax error.
Query or info() return the promise but no callback

**Comment by finchen** [comment]
I was somehow working with pouchdb 4.0.7
I have upgraded to 5.0.0 and it is now clear that it is a replication error when trying to get an attachment.

replicate() ends in 'error' with *Database has encountered an unknow error* name = unknown error 500

```
Replication started
192.168.1.4:28pouchdb:api http://townapp.nz:5984/waihibeach_test +414ms allDocs
Object
192.168.1.4:28pouchdb:api http://townapp.nz:5984/waihibeach_test +601ms allDocs success
Object
192.168.1.4:28pouchdb:api http://townapp.nz:5984/waihibeach_test +2ms bulkGet
Object
192.168.1.4:28pouchdb:api http://townapp.nz:5984/waihibeach_test +411ms bulkGet error
PouchError
http://xxx:5984/test/_bulk_get?revs=true&attachments=true&_nonce=1446326127079Failed to load resource
192.168.1.4:28Replication error
PouchError
```

**Comment by finchen** [comment]
while my initial bug report was for a query. it seems now that it is a replication problem

pouchdb: 4.0.7
couchdb: 1.6.0 with one document created with couched web administration

```// local sqlite db (cordova plugin 0.7.11)
var db = new PouchDB('test', {adapter: 'websql'});
// remote 20mb database
var remoteDB = new PouchDB("http://xxxxx:5984/mydb",{ajax: {timeout: 60000}});

// replication is a success
db.replicate.from(remoteDB)
.on('active', function () {
     console.log('Replication started');
})
on('complete', function (res) {
    console.log("DB synchronized!", res);
})
.on('error', function (err) {
        console.log('Replication error', err);
});
```

```
Replication started
192.168.1.4:28 pouchdb:api http://xxx:5984/waihibeach_test +414ms allDocs
192.168.1.4:28 pouchdb:api http://xxx:5984/waihibeach_test +601ms allDocs success
192.168.1.4:28 pouchdb:api http://xxx:5984/waihibeach_test +2ms bulkGet
192.168.1.4:28 pouchdb:api http://xxx:5984/waihibeach_test +411ms bulkGet error
PouchError http://xxx:5984/test/_bulk_get?revs=true&attachments=true&_nonce=1446326127079 Failed to load resource
192.168.1.4:28 Replication error
PouchError
```

did it work sometimes..? yes
I have seen replication successful but fail on compat()
or a complete success and fail on the next replication.
I know.. it won't help you ..

I will try later to reset everything and try again

**Comment by ronyrun** [comment]
This error is normal.
_bulk_get is not yet supported but will be soon.

**Comment by nolanlawson** [comment]
> if I remove the sqlite plugin, it is working. 

The SQLite Plugin does not fully pass the PouchDB test suite, so we don't recommend it anymore. Your best bet is normal WebSQL/IndexedDB or the Fruitdown adapter (for WKWebView).

**Comment by finchen** [comment]
That's a shame..

In this case you should change http://pouchdb.com/adapters.html
and make it clear that there is no way to use the replication on iOS for a database > 50mb.

When loading the database with pouchdb-load it seems to work fine.

**Comment by agozie** [comment]
@finchen Shame for all the team's good and FREE work? Go fix it yourself!

**Comment by finchen** [comment]
'shame / pity' the sqlite plugin does not pass the test suite.. 
it seems so close to work. 
I am not skilled enough to fix it myself

**Comment by daleharvey** [comment]
The replication error is filed and will be fixed in https://github.com/pouchdb/pouchdb/issues/4542, There is no need for any hostile messsage :)

**Comment by finchen** [comment]
just for information.
I am back trying to use iOS and the cordova-plugin-sqlite-2 
I got the same error "SyntaxError: Unexpected EOF http://192.168.1.4/#/home" at the replication

I used a small node script to look for \u2028 character in my docs
`var db       = require('nano')('http://localhost.nz:5984/mydb')
  , params   = {include_docs: true, descending: true}
  ;

db.list(params, function(error,body,headers) {
  //console.log(body);

  for (i = 0; i < body.rows.length; i++) {
    //console.log(body.rows[i].doc);
    var doc = body.rows[i].doc;

    for (var prop in doc) {
    	if( typeof doc[prop] === 'string' ){
    		if( doc[prop].indexOf("\u2028") !== -1 || doc[prop].indexOf("\\u2028") !== -1 ){
    			console.log(doc[prop].indexOf("\u2028"), doc._id, doc.name, prop);
    		}
    	}
	}
  }
});`


one of my document had a \u2028 character in a field. After removing it, the replication is working now perfectly.
Thanks
